components/FoodGroups/Draggable/characterspawner.tsx

```diff
@@ -0,0 +1,243 @@
+import { useEffect, useState } from 'react'
+
+import Draggable, { DRAGGING_STATE } from '@components/FoodGroups/Draggable'
+import {
+  ORIGIN_VECTOR2,
+  Vector2
+} from '@components/FoodGroups/Draggable/boundingbox'
+import { FOOD_GROUPS, GROUPS } from '@components/FoodGroups/groups'
+
+import {
+  FoodGroupCharacterImage,
+  FoodGroupCharacterImageDynamic
+} from './types'
+import { State_, StateDispatch } from '../types'
+
+const N_DRAGGABLES = 5
+
+const updatedFunction = (f: Function) => {
+  const [rerender, setRerender] = useState(0)
+  useEffect(() => {
+    f()
+  }, [rerender])
+  return () => setRerender(rerender + 1)
+}
```

Comment:
Only lint error left is to do with this function. I'm unsure what the rerender state is for tbh. Seems like it only gets created locally in the function so doesn't it get set back to 0 each call?

Reply:
this function is a big hack to force a rerender, to make sure the draggable "knows" what the segment the character is dropped on. will look into seeing how we can fix it again

Reply:
Used ref to fix this workaround, resolved